tutorial/01_Introduction/intro.ipynb

```diff
@@ -8,7 +8,7 @@
    "source": [
```

Comment:
<p>maybe explain a bit more what exactly <code>finalize</code> does and if it is needed also if running serially.</p><p> </p><p>Also if anyone forgets to run it, are there warnings or errors thrown?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/hpc4cmb/toast/pull/369/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='74'/>

Reply:
<p>Good points.  Some operators don't even need or have a <code>finalize()</code>, but for others it is a critical feature that could run expensive operations after accumulating timestream data across many <code>exec()</code> calls.  Perhaps for interactive use like this notebook it would be OK to define a higher-level method (<code>apply()</code>?)  that under the hood just calls exec and then finalize.  The Pipeline operator would still use the lower-level methods to do things in the correct sequence.</p>

Reply:
<p>The notebook now talks more about <code>exec()</code>,  <code>finalize()</code>,  and the new method <code>apply()</code>,  which does a single call to exec followed by finalize.  Using the <code>apply()</code>  method is convenient for serial operations processing all detectors at once.</p>